plugins/modules/ec2_asg_instance_refresh.py

```diff
@@ -0,0 +1,205 @@
+version_added: 3.2.0
+short_description: Start or cancel an EC2 Auto Scaling Group (ASG) instance refresh in AWS
+description:
+  - Start or cancel an EC2 Auto Scaling Group instance refresh in AWS.
+  - Can be used with ec2_asg_instance_refreshes_info to track the subsequent progress
+requirements: [ boto3 ]
+author: "Dan Khersonsky (@danquixote)"
+options:
+  state:
+    description:
+      - Desired state of the ASG
+    type: str
+    required: true
+    choices: [ 'started', 'cancelled' ]
+  name:
+    description:
+      - The name of the auto scaling group you are searching for.
+    type: str
+    required: true
+  strategy:
+    description:
+      - The strategy to use for the instance refresh. The only valid value is Rolling.
+      - A rolling update is an update that is applied to all instances in an Auto Scaling group until all instances have been updated.
+      - A rolling update can fail due to failed health checks or if instances are on standby or are protected from scale in.
+      - If the rolling update process fails, any instances that were already replaced are not rolled back to their previous configuration.
+    type: str
+    default: 'Rolling'
+  preferences:
+    description:
+      - Set of preferences associated with the instance refresh request.
+      - If not provided, the default values are used.
+      - For I(min_healthy_percentage), the default value is C(90).
+      - For InstanceWarmup, the default is to use the value specified for the health check grace period for the Auto Scaling group.
+    required: false
+    suboptions:
+      min_healthy_percentage:
+        description:
+          - The amount of capacity in the Auto Scaling group that must remain healthy during an instance refresh to allow the operation to continue,
+          - as a percentage of the desired capacity of the Auto Scaling group (rounded up to the nearest integer).
+          - The default is 90.
+        type: int
+      instance_warmup:
+        description:
+          - The number of seconds until a newly launched instance is configured and ready to use.
+          - During this time, Amazon EC2 Auto Scaling does not immediately move on to the next replacement.
+          - The default is to use the value for the health check grace period defined for the group.
+        type: int
+    type: dict
+extends_documentation_fragment:
+- amazon.aws.aws
+- amazon.aws.ec2
+
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+- name: Start a refresh
+  community.aws.ec2_asg_instance_refresh:
+    name: some-asg
+    state: started
+
+- name: Cancel a refresh
+  community.aws.ec2_asg_instance_refresh:
+    name: some-asg
+    state: cancelled
+
+- name: Start a refresh and pass preferences
+  community.aws.ec2_asg_instance_refresh:
+    name: some-asg
+    state: started
+    preferences:
+      min_healthy_percentage: 91
+      instance_warmup: 60
+
+'''
+
+RETURN = '''
```

Comment:
I guess the RETURN block of ec2_asg_instance_refresh should be consistent with ec2_asg_instance_refreshes_info.

Reply:
based on API documentation, the return block seems correct to me.
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/autoscaling.html#AutoScaling.Client.start_instance_refresh
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/autoscaling.html#AutoScaling.Client.cancel_instance_refresh

Please let me know if there is anything that I might be missing?

Reply:
IMO it should has the same return block as the _info module, but, not sure what @marknet15 or @markuman suggest.

Reply:
Yep I guess generally it would seem logical to me that the return would be near identical to that returned from the info module, so yep it would make sense to document that